src/libraries/Common/src/System/Text/ValueStringBuilder.cs

```diff
@@ -243,7 +246,8 @@ public unsafe void Append(char* value, int length)
             Span<char> dst = _chars.Slice(_pos, length);
             for (int i = 0; i < dst.Length; i++)
             {
-                dst[i] = *value++;
+                dst[i] = value;
+                value = ref Unsafe.Add(ref value, 1);
```

Comment:
Looks like you'll need a PackageReference in a place where you now use this https://dev.azure.com/dnceng/public/_build/results?buildId=1636207&view=logs&j=81a0f8c2-d4db-561f-9b8e-3f045280d7fe&t=2a104984-c43b-5b93-1361-6bd8a1e96b04&l=1991